### DISCUSSION TITLE
Resize the main window

### DISCUSSION THREAD

**Discussion by mtp** [AllenDang/giu#128] [Q&A] [answered]
Hello everyone,

Do you know if it's possible to resize the main window ?

Thanks,

**Comment by mtp** [comment, marked as answer]
Si I looked at the MasterWindow.go file.
I found the following function 
```go
// Set position of master window.
func (w *MasterWindow) SetPos(x, y int) {
	if w.platform != nil {
		if glfwPlatform, ok := w.platform.(*imgui.GLFW); ok {
			glfwPlatform.GetWindow().SetPos(x, y)
		}
	}
}
```

So I used this one to create a SetSize function as :
```go
// Set size of master window.
func (w *MasterWindow) SetSize(width, height int) {
	if w.platform != nil {
		if glfwPlatform, ok := w.platform.(*imgui.GLFW); ok {
			glfwPlatform.GetWindow().SetSize(width, height)
		}
	}
}
```

But when I call this function when the app is already running, my program is exiting with the following error :
```
Assertion failed!

Program: C:\xxx\project\__debug_bin
File: imgui.cpp, Line 6889
```

**Reply by AllenDang** [reply]
A quick think to me is to read the GLFW's document and check the invoke condition of SetSize. I'll check this out also when I get back to my laptop.

**Reply by mtp** [reply]
Thanks @AllenDang for your answer.

I've checked, and it's really the same instructions as for SetPos.
https://github.com/go-gl/glfw/blob/master/v3.3/glfw/window.go#L495

SetPos is working, SetSize crashs my app.
Even if I call the SetSize from a mainthread.Call, it give me a deadlock...

**Reply by AllenDang** [reply]
@mtp I added SetSize to MasterWindow and it was tested should work at least on MacOS. We need to call SetSize within mainthread.CallNonBlock.

**Reply by mtp** [reply]
It perfectly solved the issue, thanks @AllenDang
